samples/nrf9160/modem_shell/src/uart/uart_shell.c

```diff
@@ -34,15 +36,42 @@ static int print_help(const struct shell *shell, size_t argc, char **argv)
 	return ret;
 }
 
+static void uart_disable_handler(struct k_work *work)
+{
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
+	int err = nrf_modem_lib_trace_stop();
+
+	if (err) {
+		mosh_print("nrf_modem_lib_trace_stop failed with err = %d.", err);
+		return;
+	}
+#endif
+	disable_uarts();
+}
+
+static void uart_enable_handler(struct k_work *work)
+{
+	enable_uarts();
+	mosh_print("UARTs enabled\n");
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
+	int err = nrf_modem_lib_trace_start(NRF_MODEM_LIB_TRACE_ALL);
+
+	if (err) {
+		mosh_print("nrf_modem_lib_trace_start failed with err = %d.", err);
+	}
+#endif
+}
+
+static K_WORK_DEFINE(uart_disable_work_q, &uart_disable_handler);
```

Comment:
This is not a workqueue, so "_q" should probably be removed.
```suggestion
static K_WORK_DEFINE(uart_disable_work, &uart_disable_handler);
```